Tasks/Common/nuget-task-common/NuGetXmlHelper.ts

```diff
@@ -0,0 +1,128 @@
+                let xmlPackageSources = NuGetXmlHelper._getLastElement(xml, "packageSources");
+                let xmlSource = xmlPackageSources.c("add", {
+                    key: name,
+                    value: source
+                });
+
+                if (username || password) {
+                    if (!username || !password) {
+                        tl.debug('Adding NuGet source with username and password, but one of them is missing.');
+                    }
+
+                    xml = this._addCredentialsToSource(xml, name, username, password);
+                }
+            }
+
+            return xml;
+        });
+    }
+
+    public RemoveSourceFromNuGetConfig(name: string): void {
+        NuGetXmlHelper._updateXmlFile(this._nugetConfigPath, (xml: any): any => {
+            if (xml) {
+                var xmlSources = xml.getChildrenByFilter((child: any): boolean => {
+                    return typeof(child) === 'object' &&
+                           child.getName().toLowerCase() === "add" &&
+                           child.up().getName().toLowerCase() === "packagesources" &&
+                           child.attrs.key === name;
+                }, true);
+
+                xmlSources.forEach((xmlSource: any): void => {
+                    xmlSource.up().remove(xmlSource);
+                });
+
+                xml = this._removeSourceCredentials(xml, name);
+            }
+
+            return xml;
+        });
+    }
+
+    private _addCredentialsToSource(xml: any, name: string, username: string, password: string): any {
+        if (xml) {
+            let xmlSourceCredentials = NuGetXmlHelper._getLastElement(xml, "packageSourceCredentials");
+            let xmlFeedName = xmlSourceCredentials.c(NuGetXmlHelper._nuGetEncodeElementName(name));
+            xmlFeedName.c("add", {
+                key: "Username",
+                value: username
+            });
+            xmlFeedName.c("add", {
+                key: "ClearTextPassword",
+                value: password
+            });
+        }
+
+        return xml;
+    }
+
+    private _removeSourceCredentials(xml: any, name: string): any {
+        if (xml) {
+            let xmlSourceCredentials = xml.getChildrenByFilter((child: any): boolean => {
+                return typeof(child) === 'object' &&
+                       child.getName() === NuGetXmlHelper._nuGetEncodeElementName(name) &&
+                       child.up().getName().toLowerCase() === "packagesourcecredentials";
+            }, true);
+
+            xmlSourceCredentials.forEach((xmlCredentials) => {
+                xmlCredentials.up().remove(xmlCredentials);
+            });
+        }
+
+        return xml;
+    }
+
+    private static _getLastElement(xml: any, elementName: string): any {
+        if (xml) {
+            let xmlElements = xml.getChildren(elementName);
+            if (!xmlElements || xmlElements.length === 0) {
+                xmlElements = [xml.c(elementName)];
```

Comment:
Was a bit confused here, until I realized that xml.c will insert this new element just inside the root, which is the correct place for "packageSources" and "packageSourceCredentials".
Maybe a comment here, maybe we should throw if the element name doesn't match one of these two to make it clear what we can legally insert.